ui/src/main/java/be/appwise/ui/EditText.kt

```diff
@@ -0,0 +1,730 @@
+ *
+ * Slider using custom track and thumb:
+ *
+ * @param value current value of the slider. If outside of [valueRange] provided, value will be
+ * coerced to this range.
+ * @param label the optional label to be displayed above the text field container.
+ * @param onValueChange callback in which value should be updated
+ * @param modifier the [Modifier] to be applied to this slider
+ * @param enabled controls the enabled state of this slider. When `false`, this component will not
+ * respond to user input, and it will appear visually disabled and disabled to accessibility
+ * services.
+ * @param valueRange range of values that this slider can take. The passed [value] will be coerced
+ * to this range.
+ * @param colors [SliderColors] that will be used to resolve the colors used for this slider in
+ * different states. See [SliderDefaults.colors].
+ * @param interactionSource the [MutableInteractionSource] representing the stream of [Interaction]s
+ * for this slider. You can create and pass in your own `remember`ed instance to observe
+ * [Interaction]s and customize the appearance / behavior of this slider in different states.
+ * @param thumb the thumb to be displayed on the slider, it is placed on top of the track. It takes a [Brush] so its more customizable
+ * @param icon the optional icon to be displayed in front of the message or error message. Default value is Icons.Outlined.ErrorOutLine.
+ * @param message the optional error/info message. The message is shown underneath the text field,
+ * when isError is true this message will be an error message and shown in ErrorColor from the appTheme.
+ * @param isRequired when `true`, the input will show a * behind the label in the Red color.
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun EditTextSlider(
+    modifier: Modifier = Modifier,
+    value: Float = 0f,
+    onValueChange: (Float) -> Unit = {},
+    label: String,
+    enabled: Boolean = true,
+    message: String = "",
+    valueRange: ClosedFloatingPointRange<Float> = 0f..1f,
+    icon: ImageVector = Icons.Outlined.ErrorOutline,
+    isRequired: Boolean = false,
+    colors: SliderColors = SliderDefaults.colors(
+        thumbColor = Color.Transparent,
+        disabledThumbColor = Color.Transparent,
+        activeTickColor = Color.Transparent,
+        inactiveTickColor = Color.Transparent,
+        activeTrackColor = MaterialTheme.colorScheme.primary
+    ),
+    thumb: Brush = Brush.radialGradient(
+        0.6f to Color.White,
+        0.6f to MaterialTheme.colorScheme.primary,
+        1f to MaterialTheme.colorScheme.primary,
+    ),
+    interactionSource: MutableInteractionSource = remember { MutableInteractionSource() }
+) {
+
+    Column {
+        Label(label = label, isRequired = isRequired)
+
+        Box(
+            modifier = Modifier
+                .fillMaxWidth()
+                .background(color = Color.Gray, shape = RoundedCornerShape(10.dp)),
```

Comment:
I would move this backgroundColor + shape to the parameter list to be able to customise it better